en/docs/install-and-setup/setup/security/logins-and-passwords/harshicrop-vault-extension.md

```diff
@@ -0,0 +1,298 @@
+   `<APIM_HOME>/repository/components/lib/` directory.
+
+4. Create the `/repository/conf/security/secret-conf.properties` file and set the following configurations:
+
+    ```properties
+    keystore.identity.location=repository/resources/security/wso2carbon.jks
+    keystore.identity.type=JKS
+    keystore.identity.store.password=identity.store.password
+    keystore.identity.store.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    keystore.identity.key.password=identity.key.password
+    keystore.identity.key.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    carbon.secretProvider=org.wso2.securevault.secret.handler.SecretManagerSecretCallbackHandler
+
+    secVault.enabled=true
+    secretRepositories=vault
+    secretRepositories.vault.provider=org.wso2.carbon.securevault.hashicorp.repository.HashiCorpSecretRepositoryProvider
+    secretRepositories.vault.properties.address=http://127.0.0.1:8200
+    secretRepositories.vault.properties.namespace=ns1
+    secretRepositories.vault.properties.enginePath=wso2apim
+    secretRepositories.vault.properties.engineVersion=2
+    secretRepositories.vault.properties.authType=APP_ROLE
+    secretRepositories.vault.properties.roleId=<role id>
+    ```
+    !!! note
+        In production, you should always use the vault address with TLS enabled.
+
+5. Add the following lines to the `<APIM_HOME>/repository/conf/log4j2.properties` file:
+
+    ```properties
+    logger.org-wso2-carbon-securevault-hashicorp.name=org.wso2.carbon.securevault.hashicorp
+    logger.org-wso2-carbon-securevault-hashicorp.level=INFO
+    logger.org-wso2-carbon-securevault-hashicorp.additivity=false
+    logger.org-wso2-carbon-securevault-hashicorp.appenderRef.CARBON_CONSOLE.ref = CARBON_CONSOLE
+    ```
+
+    Then append `org-wso2-carbon-securevault-hashicorp` to the `loggers` list in the same file as follows:
+
+        ```properties
+        loggers = AUDIT_LOG, trace-messages, ..., org-wso2-carbon-securevault-hashicorp
+        ```
+
+### Step 3: Update passwords with their aliases
+
+1. Open the `deployment.toml` file in the `<APIM_HOME>/repository/conf/` directory and add
+   the `[secrets]` configuration section **at the bottom of the file** as shown below.
+   Give an alias for the passwords and put the value as blank (`""`).
+
+    ```toml
+    [secrets]
+    admin_password = ""
+    ```
+
+2. Add the encrypted password alias to the relevant sections in the `deployment.toml`
+   file by using a placeholder: `$secret{alias}`. For example:
+
+    ```toml
+    [super_admin]
+    username="admin"
+    password="$secret{admin_password}"
+    ```
+
+### Step 4: Start the server
+
+
+1. Provide the `Secret ID` to the prompted message in the console or by creating a new file in the `<APIM_HOME>` 
+   directory. The file should be named according to your Operating System.
+
+    === "Linux"
+        ```
+        The file name should be `hashicorpSecretId-tmp`.
```

Comment:
Let's change the formatting as suitable for a statement.

Reply:
corrected in 1d8bdffa01d44e5da79703dd2060f0a621bb659a